R/expand_model_out_grid.R

```diff
@@ -91,14 +94,27 @@
 #'   round_id = "2022-12-26",
 #'   include_sample_ids = TRUE
 #' )
+#' # Override config compound task ID set
+#' # Create coarser compound task ID set for the first modeling task which contains
+#' # samples
+#' expand_model_out_grid(config_tasks,
+#'   round_id = "2022-12-26",
+#'   include_sample_ids = TRUE,
+#'   compound_taskid_set = list(
+#'     c("forecast_date", "target"),
+#'     NULL
+#'   )
+#' )
 expand_model_out_grid <- function(config_tasks,
```

Comment:
By trying the new parameter, I got an error message if I used:
```r
compound_taskid_set = list(
                            c("origin_date"),
                            NULL
                      )
```

I fixed it by changing line 417 of this file:
from ` spl <- unique(spl[, comp_tids]) %>%` to
  `spl <- unique(spl[, comp_tids, FALSE]) %>%`, to be sure to keep the data frame output format

Reply:
will it work if I use an empty set for this list?

Reply:
I will work if you set it to `NULL`

Reply:
It does work if `compound_taskid_set` in `NULL`. This is the default signaling to use the compound task Id set in the config.

It also works if one of the list elements in `NULL`. However the question made me think of what we may want out of that scenario. Whne the question was asked, providing a list with `NULL` for a given modeling task fell back to using the config compound task id set. However, after some thought, it felt like `NULL` in this situation should replicate the behaviour when `compound_taksid_set` is set to `NULL` for a given modeling task, i.e. use all task ids in `compound_taksid_set`. So I've gone ahead and implemented the above fix in ec42ae72d187196548acfee12a869112164d2da3.